### PR TITLE
Add Popup

### PR DESCRIPTION
**Pull request by Eggiverse**
Hi again.

I add support for popup.

Be careful before merging because I had to add a `SyncButton` to make `Popup` work.
Normal `Button` cannot be used to open a `Popup` since its click event runs asynchronously.

Let me know if there is a better solution.

### CODE REVIEW

**Comment by Eggiverse** [comment]
If this is good enough, I hope to have a new release. I need to use for my project.

**Comment by wookay** [comment]
thanks. could you make `Button` like this?   so it's not need the`SyncButton`.

```julia
function imgui_control_item(imctx::Ptr, item::Button)
    CImGui.Button(item.title) && Mouse.leftClick(item)
end
```

File: Project.toml [hunk 1]
```diff
@@ -4,33 +4,33 @@ keywords = ["GUI"]
 license = "MIT"
 desc = "GUI programming in Julia based on CImGui.jl"
 authors = ["WooKyoung Noh <wookay.noh@gmail.com>"]
```
Comment (wookay) [inline Project.toml:6]: I want to add you to the authors.

**Comment by Eggiverse** [comment]
> thanks. could you make `Button` like this? so it's not need the`SyncButton`.
> 
> ```julia
> function imgui_control_item(imctx::Ptr, item::Button)
>     CImGui.Button(item.title) && Mouse.leftClick(item)
> end
> ```

In most cases, we want the event of button click to be asynchronous so that it won't freeze GUI.
However, it cannot be asynchronous if the button click is meant to modify GUI itself.
Thus, I think `Button` should keep its current behavior.

I might be wrong on this, though, since I am not fully aware how CImgui works.

File: Project.toml (hunk 1 above)
Comment (Eggiverse) [reply Project.toml:6]: That's very kind of you to do so.

I think the easier way (or lazier way) is

```toml
authors = ["WooKyoung Noh <wookay.noh@gmail.com>", "Other contributors"]
```

😂😁

Comment (Eggiverse) [reply Project.toml:6]: I am trying to add GUI for [Comoniconjl.](https://github.com/Roger-luo/Comonicon.jl).
So far, CImGUI.jl is a mature GUI for Julia but supplies C style APIs.
I don't like writing in pure Julia but feeling writing in two languages.

When I saw your Poptart, I realized it is possible to make API nicer, 
so, I began to explore possibilities following the ideas of Poptart.

**Comment by wookay** [comment]
I think that we can use `@async` inside of the callback for the asynchronous jobs.
that's for the `CImGui.Button`, to maintain the consistency with CImGui users.
```julia
didClick(button) do event
    @async for i in 1:3
        @info :i i
    end
    @info :didClick event
end
```

another option is adding a field and passing it with kwargs something like.
```julia
Button(title="Hello", async=true)  
```

**Comment by Eggiverse** [comment]
I am thinking about this: 

```julia
@kwdef mutable struct Button{AS} <: UIControl where {AS <: Union{Val{true}, Val{false}}}
    title::String = "Button"
    callback::Union{Nothing,Function} = nothing
end

function Button(; title::String = "Button", callback::Union{Nothing,Function} = nothing, async::Bool = true)
    Button{Val(async)}(title = title, callback = callback)
end


function imgui_control_item(imctx::Ptr, item::Button{Val{true}})
    CImGui.Button(item.title) && @async Mouse.leftClick(item)
end

function imgui_control_item(imctx::Ptr, item::Button{Val{false}})
    CImGui.Button(item.title) && Mouse.leftClick(item)
end
```

I haven't tested it, but I find it a better (Julian) solution if it works.

**Comment by wookay** [comment]
that type parameter looks bad. just make a field with `async::Bool=false`.

**Comment by Eggiverse** [comment]
You are right. Type parameter  is not even working.

I'll just add field.

**Comment by Eggiverse** [comment]
I have updated this pull request.

Two problems remain:

1. I am not sure where to put function `open_popup` best and whether it a good name.
1. `Mouse.leftClick` doesn't work with `open_popup`. I am not sure why that is the case.

**Comment by wookay** [comment]
1. create `popups.jl` file under `Desktop` directory.  and I prefer the function name as `OpenPopup` not `open_popup`.

2. look at `Desktop/application.jl`
```julia
function create_app(app::UIApplication)
    glwin, imctx, glsl_version = create_window(app)
    ...
    custom_fonts(app)
    task = @async runloop(glwin, imctx, glsl_version, setup_app, app)
    app.task = task
    ...
end
```
so the app is running in async.
you can't call CImGui things in another task.
```julia
julia> app.task
Task (runnable) @0x000000011210c490

julia> current_task()
Task (runnable) @0x0000000110180010
```

**Comment by Eggiverse** [comment]
I think `Popup` is good enough to work now.

File: src/Desktop/imgui_controls.jl [hunk 2]
```diff
@@ -68,7 +68,14 @@ function imgui_control_item(imctx::Ptr, item::Slider)
 end
 
 function imgui_control_item(imctx::Ptr, item::Button)
-    CImGui.Button(item.title) && @async Mouse.leftClick(item)
+    CImGui.Button(item.title) || return
+    if item.async
+        @info "async pressed"
```
Comment (wookay) [inline src/Desktop/imgui_controls.jl:73]: it doesn't need `@info` here

**Pull request merged**